### ISSUE TITLE
No support for flutter 1.20

### ISSUE THREAD

**Issue by resheilA** [fluttercommunity/flutter_workmanager#188]
I have tried debugging my application. But as soon as i include workmanager in my pubspec.yaml it can not compile.  This was working fine before the update.

FAILURE: Build completed with 2 failures.

1: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':app:mergeDebugJavaResource'.
> A failure occurred while executing com.android.build.gradle.internal.tasks.Workers$ActionFacade
   > File 'com.android.builder.files.ZipCentralDirectory@229b9794' was deleted, but previous version not found in cache

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
==============================================================================

2: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':app:mergeDebugNativeLibs'.
> A failure occurred while executing com.android.build.gradle.internal.tasks.Workers$ActionFacade
   > File 'com.android.builder.files.ZipCentralDirectory@313e9a9e' was deleted, but previous version not found in cache

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

**Comment by gebsl** [comment]
Flutter 1.20 is still in development. Latest stable Flutter is 1.17.5.
I think you just can not expect plugins to have full compatibility with possibly unstable platform versions.

Try again with stable Flutter, if error still persists, you should provide more debug information and fill in the bug report template.

**Comment by resheilA** [comment]
No with stable version it works fine. Flutter 1.17.5 it works completely fine. I was just keen asking if there are any hacks to make it run in flutter v1.20.

**Comment by ollydixon** [comment]
@gebsl v1.20 is released now, there's even v1.20.1 now. Can't use this lib until that's fixed. v1.20 should actually make it much easier to install this lib, i.e no manual configurations.

**Comment by gebsl** [comment]
@ollydixon: Just upgraded to latest flutter version (1.20.2) and also to latest version of plugin (0.2.3) and I can not reproduce your issue. Even ran `flutter clean` to remove all build artifacts, but app is still building and background tasks still executing.

Please provide more debug information.

**Comment by ajbeattie** [comment]
@gebsl the issue seems to only exist on iOS. I'm able to build and run fine for Android. (I am on flutter 1.20.2)

On iOS, when attempting to `import workmanager` in `AppDelegate.swift` (to register the plugin and register other plugins needed for background work), I get a swift compile error in xcode: `No such module 'workmanager'`. If I remove this from `AppDelegate.swift`, I get a `MissingPluginException` when the app builds and attempts to use workmanager.

Flutter doctor finds no issues, and I've attempted to `clean`, `pub get`, and remove podfile/`pod install`, all without success.

**Comment by gebsl** [comment]
@ajbeattie: Alright, thanks for the information.

At the moment I'm abroad and don't have my MacBook with me. I will try to build my iOS app next week and see, if I will face the same error. But to be honest: I'm not very used iOS development, so don't expect some sophisticated from me ;)

**Comment by ajbeattie** [comment]
@gebsl No problem, and thanks for the help :)

I'm also not very knowledgeable about iOS development, or I would try to crack the issue myself.

**Comment by gebsl** [comment]
Can confirm the same error (flutter 1.20.2) when building on iOS. After upgrading flutter I was informed that my Podfile was out of date (while running `flutter build ios`). I deleted my Podfile and it was recreated automatically after running `flutter build ios` the second time and I hit the exact same error. But this time, the pod `workmanager` was missing in my project, which could be the problem. Also, there is no symlink created to Workmanager's project in "ios/.symlinks", which is odd.

I did a `flutter clean`, `rm -rf ios/.symlinks`, *discarded the new Podfile* and this way I was able to build the application without any problems.

However, this seems to break this workaround -> https://github.com/fluttercommunity/flutter_workmanager/issues/189#issuecomment-663868680

I will try to build my app without this workaround (which was necessary for successful background fetch) and get back here, if there is any change in functionality.
